### DISCUSSION TITLE
automatic hyperlink from url

### DISCUSSION THREAD

**Discussion by dkebler** [squidfunk/mkdocs-material#7261] [Q&A] [answered]
Is is possible to turn on a feature for entering a url like

`https://github.com/squidfunk/mkdocs-material/discussions/categories/q-a`

so mkdocs will automatically make a hyperlink like github discussions did below (and many other markdown based sites (like stack exchange)

https://github.com/squidfunk/mkdocs-material/discussions/categories/q-a

if so can't find it in the docs

**Comment by alexvoss** [comment]
The usual answer would be that there is a plugin for this but I have not been able to find one that does this. Perhaps this is because such links are not often used in published material. Perhaps I have simply missed something and someone else has more luck. 

You could try adding a [hook](https://www.mkdocs.org/user-guide/configuration/#hooks) that attaches to the [on_page_markdown](https://www.mkdocs.org/dev-guide/plugins/#on_page_markdown) event and changes all links that are not part of the Markdown syntax into ones that are.

**Reply by dkebler** [reply]
agreed not needed for published material but I use mkdocs mostly for my own personal note taking so it's a bit of a hassle to keep having to make md hyperlink instead of copy and paste and move on. 

I'll look into what you say.  Thx.

**Reply by kamilkrzyskow** [reply]
When you see patterns in Markdown, first think about possible extensions, plugins add more overhead, but are easier to implement, so people opt into that more 😅 

I asked ChatGPT if there is a Python Markdown extension doing what you ask and it replied with:
- https://pypi.org/project/markdown-urlize/

Unfortunately, an unmaintained project with some bugs. In the https://github.com/r0wb0t/markdown-urlize/issues there are some suggestions to make it work better via modifying the source code regex.

And to top it off, it doesn't work with MkDocs because the API interface crashes during extension load, so changing the code would be a requirement, but is a good heads start.

**Reply by alexvoss** [reply]
How about that - Markdown supports "autolinks": <https://daringfireball.net/projects/markdown/syntax#autolink> - just two additional angle brackets.

**Reply by kamilkrzyskow** [reply]
Wow, great find. I didn't know this is part of the standard. 
I often wrap links with \<\> to avoid rendering social cards/embeds on Discord, yet never considered the usage outside of that 😅

I have now asked about it ChatGPT in a new thread (as a sort of sanity check), and it replied that Python Markdown support autolinks per CommonMark spec..., and when I corrected it that is not true it replied to me with yet another extension 😵 💫 
- https://github.com/daGrevis/mdx_linkify

More recent and more maintained than the previous one + it works with MkDocs, tested with `https://github.com/squidfunk/mkdocs-material/discussions/categories/q-a`

```
pip install mdx_linkify
```

```yml
markdown_extensions:
  - mdx_linkify
```

**Comment by squidfunk** [comment, marked as answer]
I'm late to the party, but from what I understand, Magiclink might be what you're looking for:
https://facelessuser.github.io/pymdown-extensions/extensions/magiclink/

It's already available in Material for MkDocs, since Python Markdown Extensions is a dependency:

``` yaml
markdown_extensions:
  - pymdownx.magiclink
```

**Reply by dkebler** [reply]
Sweet!  This works great.  Thx everyone for the input.  

@squidfunk  Is it worth putting a note for this extension in the mkdocs-material repo docs?  

If so I can do a PR

Or maybe not since now one will find answer in here in this discussion if they search.

**Reply by squidfunk** [reply]
[We talk about Python Markdown Extensions a lot](https://squidfunk.github.io/mkdocs-material/setup/extensions/python-markdown-extensions/), and we should not duplicate the entirety of their documentation 😉

**Comment by dkebler** [comment]
...so note to self (others who found their way here) :   

No love in mkdocs-material docs search, then search   

https://facelessuser.github.io/pymdown-extensions/ 

b4 asking    

search "automatic link" or "automatic url" and magiclink is second in the results...duh.   Sorry for wasting folks time.

**Reply by kamilkrzyskow** [reply]
No worries, I've once read the whole index of PyMdown extensions to be familiar with the use-cases, and you see how it panned out, 
I was sure there is no such extension there, so I didn't bother searching there just turned to GPT 😅

**Reply by alexvoss** [reply]
FWIW, I did look at the pmydownx documentation once you had mentioned extensions, @kamilkrzyskow, and still managed to miss it. It is fine since in the end we all contributed useful bits. The overall issue is that there are multiple places to look for these things. Perhaps we need a chatbot specialized for the sum of the relevant docs.

**Reply by squidfunk** [reply]
We're aware of the ranking problems of our current search solution. I recommend to subscribe to #6307 ☺️
